core/src/main/java/feign/DefaultMethodHandler.java

```diff
@@ -1,114 +1,164 @@
+
 import java.lang.invoke.MethodHandle;
 import java.lang.invoke.MethodHandles;
 import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 
+import feign.InvocationHandlerFactory.MethodHandler;
+
+
 /**
  * Handles default methods by directly invoking the default method code on the interface. The bindTo
  * method must be called on the result before invoke is called.
  */
 final class DefaultMethodHandler implements MethodHandler {
-  // Uses Java 7 MethodHandle based reflection. As default methods will only exist when
-  // run on a Java 8 JVM this will not affect use on legacy JVMs.
-  // When Feign upgrades to Java 7, remove the @IgnoreJRERequirement annotation.
-  private final MethodHandle unboundHandle;
-
-  // handle is effectively final after bindTo has been called.
-  private MethodHandle handle;
-
-  public DefaultMethodHandler(Method defaultMethod) {
-    Class<?> declaringClass = defaultMethod.getDeclaringClass();
-
-    try {
-      Lookup lookup = readLookup(declaringClass);
-      this.unboundHandle = lookup.unreflectSpecial(defaultMethod, declaringClass);
-    } catch (NoSuchFieldException | IllegalAccessException | IllegalArgumentException
-        | InvocationTargetException ex) {
-      throw new IllegalStateException(ex);
+    // Uses Java 7 MethodHandle based reflection. As default methods will only exist when
+    // run on a Java 8 JVM this will not affect use on legacy JVMs.
+    // When Feign upgrades to Java 7, remove the @IgnoreJRERequirement annotation.
+    public MethodHandle unboundHandle;
+
+    // handle is effectively final after bindTo has been called.
+    private MethodHandle handle;
+
+
+    public DefaultMethodHandler(Method defaultMethod) {
+        /**
+         * 13/06/2021
+         *
+         * Modified to work with Android. Previous commit worked up to API level 28 (Android 8)
+         * however, Android 9+ enforces more strict checking with reflection.
+         *
+         * Solution is to use double reflection, as below.
+         *
+         * Jamie (xrpdevs-at-xrpdevs-co-uk)
+         *
+         **/
+
+        Class<?> declaringClass = defaultMethod.getDeclaringClass();
+
+        try {
+            if (System.getProperty("java.vm.name").equalsIgnoreCase("Dalvik")) {
+                androidObtainHandle(defaultMethod, declaringClass);
+            } else {
+                Lookup lookup = readLookup(declaringClass);
+                this.unboundHandle = lookup.unreflectSpecial(defaultMethod, declaringClass);
+            }
+
+        } catch (IllegalAccessException ex) {
+            throw new IllegalStateException(ex);
+        } catch (InstantiationException | InvocationTargetException | NoSuchMethodException e) {
+            e.printStackTrace();
+        } catch (NoSuchFieldException e) {
+            e.printStackTrace();
```

Comment:
this is not merge-able, as this is eating the exceptions.  If we are unable to get the `unboundHandle` instance, throw the exception and let it blow.

Since you are going to touch this, please also include an message to this exception

Reply:
Currently I'm building this with Gradle in Android Studio.

Will refactor tomorrow and try to build in Maven against different JDKs.

Reply:
See latest commit. Hopefully this fixes problems you mentioned

Reply:
I will wait until build passes to give this a second look